### PR TITLE
[CARBONDATA-1719][Pre-Aggregate][Bug] Fixed bug to handle data inconsistency on concurrent data load and pre-aggregate table creation

### PR DESCRIPTION
**Pull request by SangeetaGulia**
Problem: On concurrent data load and pre-aggregate table creation, datamap was not getting populated with the load data even after data load completes.

This PR fix this issue. 

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [x] Any interfaces changed? No
 
 - [x] Any backward compatibility impacted? No
 
 - [x] Document update required? No

 - [x] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
    Tested the functionality on Two node cluster      
 
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. (N/A)

### CODE REVIEW

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateListeners.scala [hunk 1]
```diff
@@ -38,7 +40,10 @@ object LoadPostAggregateListener extends OperationEventListener {
     val loadEvent = event.asInstanceOf[LoadTablePreStatusUpdateEvent]
     val sparkSession = loadEvent.sparkSession
     val carbonLoadModel = loadEvent.carbonLoadModel
-    val table = carbonLoadModel.getCarbonDataLoadSchema.getCarbonTable
+    val metastore = CarbonEnv.getInstance(sparkSession).carbonMetastore
+    carbonLoadModel.getTableName
```
Comment (kunal642) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateListeners.scala]: i think this was added by mistake.. Please remove

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateListeners.scala [hunk 2]
```diff
@@ -38,7 +40,9 @@ object LoadPostAggregateListener extends OperationEventListener {
     val loadEvent = event.asInstanceOf[LoadTablePreStatusUpdateEvent]
     val sparkSession = loadEvent.sparkSession
     val carbonLoadModel = loadEvent.carbonLoadModel
-    val table = carbonLoadModel.getCarbonDataLoadSchema.getCarbonTable
+    val metastore = CarbonEnv.getInstance(sparkSession).carbonMetastore
+    val table = metastore.lookupRelation(Some(carbonLoadModel.getDatabaseName),
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateListeners.scala]: You can use CarbonEnv to get CarbonTable

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2322/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1109/

**Comment by SangeetaGulia** [comment]
retest this please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2554/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2327/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1117/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2336/

**Comment by jackylk** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2444/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1224/

File: integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateListeners.scala [hunk 3]
```diff
@@ -161,7 +163,8 @@ object PreAggregateDataTypeChangePreListener extends OperationEventListener {
     }
     if (carbonTable.isChildDataMap) {
       throw new UnsupportedOperationException(
-        s"Cannot change data type for columns in pre-aggregate table ${ carbonTable.getDatabaseName
+        s"Cannot change data type for columns in pre-aggregate table ${
```
Comment (kumarvishal09) [inline integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateListeners.scala]: Please avoid unnecessary changes.

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2509/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1286/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2675/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2517/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2676/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2679/

**Comment by kumarvishal09** [comment]
LGTM
